### PR TITLE
ts support and some fixes

### PR DESCRIPTION
**Pull request by clark-john**

### CODE REVIEW

**Comment by clark-john** [comment]
Screenshot 2024-03-02 153338
Eto nga pla yung ni-fix ko po, d po kase tumugma yung number of runs sa limit

File: index.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+    error: (...messages: any[]) => void;
+    warn: (...messages: any[]) => void;
+    log: (...messages: any[]) => void;
+    info: (...messages: any[]) => void;
+    debug: (...messages: any[]) => void;
 }
-const doNothing = (...messages: any) => {};
-class VoidLogger implements loggerInterface {
+
+const doNothing = (..._messages: any) => {};
```
Comment (albertdiones) [inline index.ts:11]: should this be also any[]?

File: index.ts [hunk 2]
```diff
@@ -21,58 +23,49 @@ enum StartMode {
...
-        options: {startMode?: StartMode, intervalMode?: IntervalMode} = {
-            startMode: StartMode.actionFirst,
-            intervalMode: IntervalMode.afterFinish
-        }
-    ): recursivePromise | Promise<void | null> {
+    
+    async continuous(
+        interval: number, 
+        limit: number | null, 
+        options: { startMode?: StartMode } = {}
```
Comment (albertdiones) [inline index.ts:51]: lost the defaults?

File: index.ts [hunk 3]
```diff
@@ -84,45 +77,37 @@ class Repeater {
                     }
                     return true;
                 }
-            );
-        if (options.intervalMode === IntervalMode.fixed) {
```
Comment (albertdiones) [inline index.ts:88]: why deleted?

Comment (mjohnclark) [reply index.ts:88]: baka dati pa ung mga changes
image
dito rin kse sa commit nare is wala ung `const chain = () =>` so baka mas dati pa, pati ung `options.intervalMode` wala din

Comment (albertdiones) [reply index.ts:88]: bruh

File: index.ts (hunk 2 above)
Comment (albertdiones) [reply index.ts:51]: https://github.com/albertdiones/add_repeater/commit/99dc4d0ee336660f7a2628596ded91bf0cc15d27

Comment (albertdiones) [reply index.ts:51]: https://github.com/albertdiones/add_repeater/blob/27abe0f0cfb0eff8c5a4ad586eac5bee2c2fae69/index.ts

**Pull request merged**